actors/builtin/miner/miner_actor.go

```diff
@@ -1084,7 +1116,7 @@ func computeFaultsFromMissingPoSts(st *State, deadlines *Deadlines, beforeDeadli
 
 	deadlineFirstPartition := uint64(0)
 	var fGroups, rGroups []*abi.BitField
-	for dlIdx := uint64(0); dlIdx < beforeDeadline; dlIdx++ {
+	for dlIdx := sinceDeadline; dlIdx < beforeDeadline; dlIdx++ {
```

Comment:
This is the heart of the fix.

Reply:
Gah, this was buggy. `deadlineFirstPartition` is initialized above assuming this loop will iterate all partitions.